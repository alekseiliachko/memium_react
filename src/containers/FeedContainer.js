import { Feed } from "../components/Feed";
import { connect } from "react-redux";
import {
  loadFeed,
  likePost,
  loadLikedList,
  deleteFromLiked,
  addToSubs,
  deleteFromSubs,
  loadSubs,
  loadCategory,
} from "../redux/user/actions";

const mapStateToProps = (store) => ({
  feedLoadingStatus: store.userReducer.feedLoadingStatus,
  feed: store.userReducer.feed,
  likedList: store.userReducer.likedList,
  likedLoadingStatus: store.userReducer.likedListLoadingStatus,
  subsLoadingStatus: store.userReducer.subsLoadingStatus,
  subs: store.userReducer.subs,
});

const mapDispatchToProps = (dispatch) => ({
  loadFeed: () => dispatch(loadFeed()),
  likePost: (articleId) => dispatch(likePost(articleId)),
  loadLiked: () => dispatch(loadLikedList()),
  unLikePost: (articleId) => dispatch(deleteFromLiked(articleId)),
  addToSubs: (accId) => dispatch(addToSubs(accId)),
  deleteFromSubs: (accId) => dispatch(deleteFromSubs(accId)),
  loadSubs: () => dispatch(loadSubs()),
  loadCategory: (category) => dispatch(loadCategory(category)),
});

export const FeedContainer = connect(mapStateToProps, mapDispatchToProps)(Feed);
