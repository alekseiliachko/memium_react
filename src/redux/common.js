export const LOADING_STATUS = {
  NOT_LOADED: "NOT_LOADED",
  LOADING: "LOADING",
  LOADED: "LOADED",
};
