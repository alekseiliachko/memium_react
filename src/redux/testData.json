{
  "authReducer": {
    "username": "alex",
    "authState": "SUCCESS"
  },
  "userReducer": {
    "avatar": "blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da",
    "blackList": [
      {
        "accountId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "username": "bob",
        "name": null,
        "bio": "populated",
        "imageData": null
      }
    ],
    "details": {
      "accountId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
      "name": "Круто чел",
      "bio": "populated",
      "gender": "gender",
      "dob": 1615046987793
    },
    "subs": [
      {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "username": "zilber",
        "name": null,
        "bio": "populated",
        "imageData": null
      }
    ],
    "likedList": [
      {
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Русские мемчанские",
        "description": null,
        "date": 1615046988090,
        "category": "Anime"
      },
      {
        "id": "d3268ef9-bb52-4967-b1ee-14826b9d8a18",
        "imageUrl": null,
        "authorId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "title": "Chem?",
        "description": null,
        "date": 1615046988090,
        "category": "Chemistry"
      },
      {
        "id": "5d8e2ff1-21bc-4fe0-936f-851407318748",
        "imageUrl": null,
        "authorId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "title": "It?",
        "description": null,
        "date": 1615046988090,
        "category": "IT"
      },
      {
        "id": "5b28e638-ef12-45a7-b629-84350cc0c92e",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Bio?",
        "description": null,
        "date": 1615047293904,
        "category": "IT"
      }
    ],
    "avatarLoadingStatus": "LOADED",
    "detailsLoadingStatus": "LOADED",
    "subsLoadingStatus": "LOADED",
    "blackListLoadingStatus": "LOADED",
    "likedListLoadingStatus": "LOADED",
    "feedLoadingStatus": "LOADING",
    "feed": {
      "articles": [],
      "accounts": null
    },
    "myArticles": [
      {
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Русские мемчанские",
        "description": null,
        "date": 1615046988090,
        "category": "Anime"
      },
      {
        "id": "5b28e638-ef12-45a7-b629-84350cc0c92e",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Bio?",
        "description": null,
        "date": 1615047293904,
        "category": "IT"
      }
    ],
    "myArticlesLoadingStatus": "LOADED"
  },
  "articleReducer": {
    "articleDataById": {
      "d3268ef9-bb52-4967-b1ee-14826b9d8a18": {
        "id": "d3268ef9-bb52-4967-b1ee-14826b9d8a18",
        "imageUrl": null,
        "description": null,
        "authorId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "title": "Chem?",
        "date": 1615046988090,
        "category": "Chemistry",
        "data": "yes, and Jews are responsible."
      },
      "5d8e2ff1-21bc-4fe0-936f-851407318748": {
        "id": "5d8e2ff1-21bc-4fe0-936f-851407318748",
        "imageUrl": null,
        "description": null,
        "authorId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "title": "It?",
        "date": 1615046988090,
        "category": "IT",
        "data": "yes, and Jews are responsible."
      },
      "9064e587-3c29-4547-a5a6-51a8149456f8": {
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "description": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Русские мемчанские",
        "date": 1615046988090,
        "category": "Anime",
        "data": "yes, and Jews are responsible."
      },
      "5b28e638-ef12-45a7-b629-84350cc0c92e": {
        "id": "5b28e638-ef12-45a7-b629-84350cc0c92e",
        "imageUrl": null,
        "description": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Bio?",
        "date": 1615047293904,
        "category": "IT",
        "data": "<div><p>Рассмотрим две <b>штуки</b></p><p>Первое:</p><p><b><br></b></p></div>"
      }
    },
    "articleIdsByAuthorName": {}
  },
  "allUsersReducer": {
    "userDataById": {
      "22bb3fe5-185c-4c87-8633-1586586506b7": {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "name": null,
        "bio": "populated",
        "gender": "gender",
        "dob": 1615046987793,
        "avatar": "blob:http://localhost:3000/595ee9e0-fbd4-4712-98e1-1fd759dca517"
      },
      "32df86b7-872d-45d5-8a51-e217f0bc973d": {
        "accountId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "name": null,
        "bio": "populated",
        "gender": "gender",
        "dob": 1615046987793,
        "avatar": "blob:http://localhost:3000/2bd315a7-60e0-41b7-b856-e48559d16d73"
      },
      "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282": {
        "accountId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "name": "Круто чел",
        "bio": "populated",
        "gender": "gender",
        "dob": 1615046987793,
        "avatar": "blob:http://localhost:3000/678d4330-3d93-4470-98f3-4619ed42c932"
      }
    }
  },
  "searchReducer": {
    "articles": [],
    "authors": []
  }
}
