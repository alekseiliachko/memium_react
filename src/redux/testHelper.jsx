import {Provider} from "react-redux";
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import React from "react";

export const getStubStoreData =  () => require('./testData.json');

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

export const withStore = (initialStore) => {
    const store = mockStore(initialStore || getStubStoreData());
    return function Wrapper({ children }) {
        return <Provider store={store}>
            {children}
        </Provider>
    }
}
