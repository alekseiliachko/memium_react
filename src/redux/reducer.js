import { combineReducers, createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { authReducer } from "./auth/reducer";
import { userReducer } from "./user/reducer";
import { articleReducer } from "./article/reducer";
import { allUsersReducer } from "./allUsers/reducer";
import { searchReducer } from "./search/reducer";

const rootState = combineReducers({
  authReducer,
  userReducer,
  articleReducer,
  allUsersReducer,
  searchReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const middleware = [thunk];
export function configureStore() {
  const store = createStore(
    rootState,
    composeEnhancers(applyMiddleware(...middleware))
  );
  return store;
}
