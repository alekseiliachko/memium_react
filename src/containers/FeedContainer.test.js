import React from "react";
import renderer from "react-test-renderer";
import { FeedContainer } from "./FeedContainer";
import { withStore } from "../redux/testHelper";
import reactRouter from "react-router";

describe("feed container", () => {
  it("renders correctly", async () => {
    const Comp = withStore();
    let tree;

    reactRouter.useLocation = jest.fn().mockReturnValue({ search: "?tab=2" });

    await renderer.act(async () => {
      tree = renderer.create(
        <Comp>
          <FeedContainer />
        </Comp>
      );
      await Promise.resolve();
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
