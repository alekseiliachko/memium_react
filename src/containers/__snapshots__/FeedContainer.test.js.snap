// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feed container renders correctly 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <div
      className="MuiTabs-root WithStyles(ForwardRef(Tabs))-root-3"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": null,
            "overflow": "hidden",
          }
        }
      >
        <div
          className="MuiTabs-flexContainer MuiTabs-centered"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root Component-root-5 MuiTab-textColorPrimary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Рекомендации
            </span>
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root Component-root-5 MuiTab-textColorPrimary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Астрология
            </span>
          </button>
          <button
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root Component-root-5 MuiTab-textColorPrimary Mui-selected"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Биология
            </span>
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root Component-root-5 MuiTab-textColorPrimary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Химия
            </span>
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root Component-root-5 MuiTab-textColorPrimary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Айти
            </span>
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root Component-root-5 MuiTab-textColorPrimary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Другое
            </span>
          </button>
        </div>
        <span
          className="PrivateTabIndicator-root-6 PrivateTabIndicator-colorSecondary-8 MuiTabs-indicator WithStyles(ForwardRef(Tabs))-indicator-4"
          style={
            Object {
              "left": 0,
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>,
  <div
    className="MuiBox-root MuiBox-root-10 feed"
  >
    <div
      className="MuiBox-root MuiBox-root-11"
    />
  </div>,
]
`;
